src/vm.c

```diff
@@ -152,24 +141,59 @@ int vm_load(vm_t *v, const char *image_path)
     size_t setupsz = (setup_sectors + 1) * 512;
     boot->hdr.vid_mode = 0xFFFF;  // VGA
     boot->hdr.type_of_loader = 0xFF;
-    boot->hdr.ramdisk_image = 0x0;
-    boot->hdr.ramdisk_size = 0x0;
     boot->hdr.loadflags |= CAN_USE_HEAP | 0x01 | KEEP_SEGMENTS;
     boot->hdr.heap_end_ptr = 0xFE00;
     boot->hdr.ext_loader_ver = 0x0;
     boot->hdr.cmd_line_ptr = 0x20000;
     memset(cmdline, 0, boot->hdr.cmdline_size);
     memcpy(cmdline, KERNEL_OPTS, sizeof(KERNEL_OPTS));
     memmove(kernel, (char *) data + setupsz, datasz - setupsz);
+
+    /* setup E820 memory map to report usable memory address ranges for initrd */
+    unsigned int idx = 0;
+    boot->e820_table[idx++] = (struct boot_e820_entry){
+        .addr = 0x0, .size = ISA_START_ADDRESS - 1, .type = E820_RAM};
```

Comment:
Add comma (`,`) right after `E820_RAM`, so that `clang-format` can indent better.